File: issue-tracker-master/src/main/java/hu/elte/alkfejl/issuetracker/model/Quantity.java
package hu.elte.alkfejl.issuetracker.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

/**
 * @author Godzsák Dávid <godzsakdavid@gmail.com>
 */
@Entity
@Table(name = "Quantity")
@Data
@AllArgsConstructor
@NoArgsConstructor
//@EqualsAndHashCode(callSuper = true)
public class Quantity extends BaseEntity {

    @Column(nullable = false)
    private int quantity;
    
    @Column(nullable = false)
    private String comment;
    
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "product_id")
    private Storage product;
    
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "depository_id")
    private Depository depository;
    
    
    public void sold(int number){
        this.quantity-=number;
    }
    
    public void renewal(int number){
        this.quantity+=number;
    }
    
    
    public void comment(String comment){
        this.comment=comment;
    }
}

File: issue-tracker-master/issue-tracker/src/app/model/quantity.ts
export class Quantity {
    id: number;
    quantity: number;
    comment: string;
    product_id: number;
    depository_id: number;

   /* constructor(id?: number, quantity?: number, comment?: string, product_id?: number, depository_id?: number) {
        this.id = id;
        this.quantity = quantity;
        this.comment = comment;
        this.product_id = product_id;
        this.depository_id = depository_id;
    }*/

    constructor(quantity?: number, comment?: string, product_id?: number, depository_id?: number) {
        this.quantity = quantity;
        this.comment = comment;
        this.product_id = product_id;
        this.depository_id = depository_id;
    }
}
